src/docfx/lib/JsonUtililty.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading;
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.Docs
+{
+    /// <summary>
+    /// Provide Utilities of Json
+    /// </summary>
+    public static class JsonUtililty
+    {
+        private static readonly ThreadLocal<JsonSerializer> s_defaultSerializer = new ThreadLocal<JsonSerializer>(
+            () => new JsonSerializer
+            {
+                NullValueHandling = NullValueHandling.Ignore,
+                ReferenceLoopHandling = ReferenceLoopHandling.Serialize,
+                Converters =
+                {
+                    new StringEnumConverter { CamelCaseText = true },
+                },
+                ContractResolver = new CamelCasePropertyNamesContractResolver(),
+            });
+
+        private static readonly JsonMergeSettings s_jsonMergeSettings = new JsonMergeSettings
+        {
+            MergeArrayHandling = MergeArrayHandling.Replace,
+        };
+
+        /// <summary>
+        /// Serialize an object to TextWriter
+        /// </summary>
+        public static void Serialize(TextWriter writer, object graph, bool isStable = false, Formatting formatting = Formatting.None, JsonSerializer serializer = null)
```

Comment:
>isStable [](start = 75, length = 8)

what does "stable" mean? do we need such concept?

Reply:
stable means stable(ordered) output, which is a config item in ops-cli,

---
In reply to: [180306534](https://github.com/dotnet/docfx/pull/2609#discussion_r180306534) [](ancestors = 180306534)

Reply:
Is this really a need? or should we always keep it stable?

---
In reply to: [180314569](https://github.com/dotnet/docfx/pull/2609#discussion_r180314569) [](ancestors = 180314569,180306534)

Reply:
stable is just for diff tools, always keep is stable will  bring additional cost


---
In reply to: [180316156](https://github.com/dotnet/docfx/pull/2609#discussion_r180316156) [](ancestors = 180316156,180314569,180306534)